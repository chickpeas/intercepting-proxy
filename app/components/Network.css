.networkMain {}

.networkContainer {
  font-size: 13px;
  background-color: #fff;
  display: flex;
}
