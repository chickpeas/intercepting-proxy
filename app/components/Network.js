// @flow
import React, { Component } from 'react';
import NetworkTable from './ui/NetworkTable';
import ActionBar from './ui/ActionBar';
import InfoPanel from './ui/InfoPanel';
import styles from './Network.css';

// TODO:
// TEST,
// ActionBar and InfoPanel should go on networkPage level,
// or have their own container
// or even remove/merge this component with network page?

type Props = {
  changeInterceptFilter: (boolean) => void,
  forwardRequest: () => void,
  dropRequest: () => void,
  network: Object,
  requests: Object,
  responses: Object,
  filter: Object,
  columns: Array
};

class Network extends Component<Props> {
  props: Props;

  constructor(props) {
    super(props);
    this.state = {
      sidePanel: false,
      selected: {}
    };
  }

  handleInterceptClick = (value: boolean) => {
    this.props.changeInterceptFilter(value);
  }
  handleForwardClick = () => {
    this.props.forwardRequest();
  }
  handleDropClick = () => {
    this.props.dropRequest();
  }
  handlePanelClick = ({ original: { requestId, responseId } }) => {
    const { requests, responses } = this.props;
    this.setState({
      sidePanel: true,
      selected: {
        requestId,
        request: requests[requestId],
        response: responses[responseId]
      }
    });
  }
  handleClosePanel = () => {
    this.setState({ sidePanel: false });
  }

  render() {
    const {
      sidePanel,
      selected,
    } = this.state;
    const {
      filter,
      requests: { pendingRequest },
      columns
    } = this.props;

    const { network: { byId, byHash }, responses, requests } = this.props;
    // TODO optimize this:
    const log = byId.map((id, index) => {
      const { requestId, responseId } = byHash[id];
      if (responses[responseId] && responses[responseId].statusCode) {
        const { statusCode, mime } = responses[responseId];
        return {
          index,
          responseId,
          requestId,
          url: requests[requestId].url,
          method: requests[requestId].method,
          statusCode,
          mime
        };
      }
      const statusCode = '';

      return {
        index,
        requestId,
        url: requests[requestId].url,
        method: requests[requestId].method,
        statusCode
      };
    });
    const Info = (<InfoPanel handleClose={this.handleClosePanel} selected={selected} />);
    return (
      <div className={styles.networkMain}>
        <ActionBar
          filter={filter}
          handleClick={this.handleInterceptClick}
          handleForwardClick={this.handleForwardClick}
          handleDropClick={this.handleDropClick}
          pendingRequest={pendingRequest}
        />
        <div className={styles.networkContainer}>
          <NetworkTable columns={columns} network={log} handleClick={this.handlePanelClick} />
          { sidePanel ? Info : null }
        </div>
      </div>
    );
  }
}

export default Network;
