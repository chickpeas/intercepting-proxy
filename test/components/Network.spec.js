import { spy } from 'sinon';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// import { Provider } from 'react-redux';
// import renderer from 'react-test-renderer';
// import { BrowserRouter as Router } from 'react-router-dom';

// import { configureStore } from '../../app/store/configureStore';
import Network from '../../app/components/Network';


Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const initialState = {
    network: {
      byId: [],
      byHash: []
    },
    requests: {},
    responses: {}
  };
  const props = {
    changeGlobalFilter: spy(),
    onRowClick: spy(),
    log: [
      {
        index: 0,
        id: '123',
        requestId: '456',
        url: 'asd.com',
        methord: 'get',
        statusCode: 200
      }
    ],
  };
  const component = shallow(<Network {...props} />);
  return {
    component,
    props
  };
}

describe('Network component', () => {
  it('should render', () => {
    const { component } = setup();
    expect(component).toMatchSnapshot();
  });

  // it('should first button should call increment', () => {
  //   const { buttons, actions } = setup();
  //   buttons.at(0).simulate('click');
  //   expect(actions.increment.called).toBe(true);
  // });

  // it('should match exact snapshot', () => {
  //   const { props } = setup();
  //   const network = (
  //     <div>
  //       <Router>
  //         <Network {...props} />
  //       </Router>
  //     </div>
  //   );
  //   const tree = renderer
  //     .create(network)
  //     .toJSON();
  //
  //   expect(tree).toMatchSnapshot();
  // });

  // it('should second button should call decrement', () => {
  //   const { buttons, actions } = setup();
  //   buttons.at(1).simulate('click');
  //   expect(actions.decrement.called).toBe(true);
  // });

  // it('should third button should call incrementIfOdd', () => {
  //   const { buttons, actions } = setup();
  //   buttons.at(2).simulate('click');
  //   expect(actions.incrementIfOdd.called).toBe(true);
  // });

  // it('should fourth button should call incrementAsync', () => {
  //   const { buttons, actions } = setup();
  //   buttons.at(3).simulate('click');
  //   expect(actions.incrementAsync.called).toBe(true);
  // });
});
