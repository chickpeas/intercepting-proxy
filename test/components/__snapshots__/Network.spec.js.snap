// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Network component should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Network
    changeGlobalFilter={[Function]}
    network={
      Object {
        "byHash": Array [],
        "byId": Array [],
      }
    }
    onRowClick={[Function]}
    requests={Object {}}
    responses={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ActionBar
          filter={undefined}
          handleClick={[Function]}
          handleDropClick={[Function]}
          handleForwardClick={[Function]}
          pendingRequest={undefined}
        />,
        <div
          className="networkContainer"
        >
          <NetworkTable
            handleClick={[Function]}
            network={Array []}
          />
        </div>,
      ],
      "className": "networkMain",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "filter": undefined,
          "handleClick": [Function],
          "handleDropClick": [Function],
          "handleForwardClick": [Function],
          "pendingRequest": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NetworkTable
              handleClick={[Function]}
              network={Array []}
            />,
            null,
          ],
          "className": "networkContainer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "handleClick": [Function],
              "network": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          null,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ActionBar
            filter={undefined}
            handleClick={[Function]}
            handleDropClick={[Function]}
            handleForwardClick={[Function]}
            pendingRequest={undefined}
          />,
          <div
            className="networkContainer"
          >
            <NetworkTable
              handleClick={[Function]}
              network={Array []}
            />
          </div>,
        ],
        "className": "networkMain",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "filter": undefined,
            "handleClick": [Function],
            "handleDropClick": [Function],
            "handleForwardClick": [Function],
            "pendingRequest": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NetworkTable
                handleClick={[Function]}
                network={Array []}
              />,
              null,
            ],
            "className": "networkContainer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "handleClick": [Function],
                "network": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            null,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
